libcxx/CMakeLists.txt

```diff
@@ -491,6 +491,15 @@ include(HandleLibcxxFlags)
 # 'config-ix' use them during feature checks. It also adds them to both
 # 'LIBCXX_COMPILE_FLAGS' and 'LIBCXX_LINK_FLAGS'
 
+# Targeting the GPU requires a clang compiler and several extra flags.
```

Comment:
Why does Clang require `-nogpulib`? Don't we already pass `-nodefaultlibs`?

Also, why `-flto`?

We should set this in the CMake caches -- we don't want to add any more target-specific flags in our CMake files, this is too messy.

Reply:
`-nogpulib` is separate, it controls whether or not the GPU toolchain pulls in external vendor libraries.

`-flto` is 100% required because GPUs have almost no backwards compatibility, so unless you want to build `libc++` over 30 times you need to use LLVM-IR + LTO to defer creating object code until it's linked into the user's application. Also ELF linking isn't supported at all for AMDGPU.

I can move this bit to the CMake cache, but `CMAKE_REQUIRED_FLAGS` might need to stay, otherwise `check_cxx_flags` will not work.